dsl/result_type.go

```diff
@@ -282,10 +261,22 @@ func CollectionOf(v interface{}, adsl ...func()) *expr.ResultTypeExpr {
 	m, ok = v.(*expr.ResultTypeExpr)
 	if !ok {
 		if id, ok := v.(string); ok {
-			if dt := expr.Root.UserType(expr.CanonicalIdentifier(id)); dt != nil {
+			// Check if a result type exists with the given type name
+			if dt := expr.Root.UserType(id); dt != nil {
 				if mt, ok := dt.(*expr.ResultTypeExpr); ok {
 					m = mt
 				}
+			} else {
+				// Check if a result type exists with the given identifier
+				id, typeName, err := mediaTypeToResultType(id)
+				if err != nil {
+					eval.ReportError("invalid result type identifier %#v in CollectionOf: %s", id, err)
+				}
+				if dt := expr.Root.UserType(typeName); dt != nil {
```

Comment:
Is it ok to do the above is err is not nil? (we don't return in the test above).

Reply:
Done